test/csharp/TestOnnxRuntimeGenAIAPI.cs

```diff
@@ -524,12 +579,19 @@ public void TestAdapters()
                     generator.GenerateNextToken();
                 }
                 using var logits = generator.GetOutput("logits");
-                Assert.Equal(ElementType.float32, logits.Type());
+                if (_useCudaModel)
+                {
+                    Assert.Equal(ElementType.float16, logits.Type());
+                    // TODO: GetData with float16?
```

Comment:
I don't think we support this in GenAI yet. I have encountered the same issue when I was trying to copy this test case to Java.

@baijumeswani Is it OK to merge this first, and handle the `GetData` with float16 later?